perses/rjmc/geometry.py

```diff
@@ -235,8 +252,48 @@ def _get_relevant_angle(self, atom1, atom2, atom3):
         atom1_angles = set(atom1.angles)
         atom2_angles = set(atom2.angles)
         atom3_angles = set(atom3.angles)
-        relevant_angle = atom1_angles.intersection(atom2_angles, atom3_angles)
-        return relevant_angle.pop()
+        relevant_angle_set = atom1_angles.intersection(atom2_angles, atom3_angles)
+        relevant_angle = relevant_angle_set.pop()
+        if type(relevant_angle.type.k) != units.Quantity:
+            relevant_angle_with_units = self._add_angle_units(relevant_angle)
+        else:
+            relevant_angle_with_units = relevant_angle
+        return relevant_angle_with_units
+
+    def _add_bond_units(self, bond):
+        """
+        Add the correct units to a harmonic bond
+
+        Arguments
+        ---------
+        bond : parmed bond object
+            The bond to get units
+
+        Returns
+        -------
+
+        """
+        bond.type.req = units.Quantity(bond.type.req, unit=units.angstrom)
+        bond.type.k = units.Quantity(bond.type.k, unit=units.kilocalorie_per_mole/units.angstrom**2)
```

Comment:
Remember that the spring constants have to be multiplied by a factor of two to convert from ParmEd to OpenMM convention!